src/Chapter4.hs

```diff
@@ -489,6 +497,18 @@ Implement the 'Applicative' instance for our 'List' type.
   type.
 -}
 
+concatLists :: List a -> List a -> List a
+concatLists Empty l = l
```

Comment:
Perfect!

You can one more case to make your solution even more efficient:

```haskell
concatLists l Empty = l
```

In that case, you won't need to walk through each element of the first list when the second one is empty and return the list instantly 🚤

Reply:
Also, one small note, in Haskell `concat` name usually has a different meaning: it inlines the list of list into a normal list:

```haskell
concat :: [[a]] -> [a]
```

When you add together something, in Haskell it is usually called `append` smth. 🙂